tests/integration/testlib/assertlib.py

```diff
@@ -53,3 +53,8 @@ def assert_absent(*ifnames):
 
     current_state = statelib.show_only(ifnames)
     assert not current_state[INTERFACES]
+
+
+def assert_all_equal(iface_macs):
+    """ Assert that all the interfaces have same mac address """
+    assert len(set(iface_macs)) == 1
```

Comment:
Probably not a problem in practice but it would be a little bit nicer to also safeguard in case iface_macs is empty, for example with `assert not macs or len(set(macs)) == 1`